thorlcr/graph/thgraphslave.cpp

```diff
@@ -586,14 +586,14 @@ unsigned __int64 CSlaveActivity::queryLocalCycles() const
                 break;
         }
     }
-    unsigned __int64 localCycles = queryTotalCycles();
-    if (localCycles < inputCycles) // not sure how/if possible, but guard against
+    unsigned __int64 processCycles = queryTotalCycles() + queryLookAheadCycles();
+    if (processCycles < inputCycles) // not sure how/if possible, but guard against
         return 0;
-    localCycles -= inputCycles;
+    processCycles -= inputCycles;
     const unsigned __int64 blockedCycles = queryBlockedCycles();
-    if (localCycles < blockedCycles)
+    if (processCycles < blockedCycles)
```

Comment:
shouldn't be possible either (worth commenting).

Perhaps we should be issuing IWARNLOG's for this and the on on 590.
Have you check if any now occur, e.g. running the regression suite?
Worth leaving _DEBUG logging in at least and running suite and seeing..

Reply:
>for this and the on on 590.

@shamser - you've added one for '(processCycles < blockedCycles)' but not for '(processCycles < inputCycles)' ?

curious if you ran the regression suite and saw any incidents?

Reply:
@shamser - now that the indexread start double counting fix is merged, are there any incidents of either "processCycles < inputCycles" or "processCycles < blockedCycles" remaining?

Unless there are many ^^ , it would still be worth logging a warning re. the "processCycles < blockedCycles" case.

Reply:
I added IWARNLOG for processCycles < inputCycles.  The regression suite produces 1646 warnings.  Should leave this warning in there? @jakesmith

Reply:
as long as it is safe to say that it is not related to changes in this PR, and given the high numbers, we should leave this off (out for now), open a new JIRA (link to this one) and investigate.